gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -296,6 +295,7 @@ public String getProgress() {
   @Override
   @SuppressWarnings("unchecked")
   public void run() {
+    recordsPulled.set(0L);
```

Comment:
For retry from the beginning, setting `recordsPulled` to 0 makes sense. But for retry from failed point, this logic is wrong.  We do need a configuration key to identify which retry mode is favored, making sure each is implemented correctly (all states should be cleared for retry from beginning, keep previous states for retry from failed point).